### PR TITLE
ref(ts) Convert TeamStore to typescript

### PR DESCRIPTION
**Pull request by markstory**
Convert to typescript and fix a null case caught by typescript.

### CODE REVIEW

File: src/sentry/static/sentry/app/stores/teamStore.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      if (response.slug !== itemId) {
+        // Remove old team
+        this.state = this.state.filter(({slug}) => slug !== itemId);
+        // Add team w/ updated slug
+        this.state.push(response);
+        this.trigger(new Set([response.slug]));
+        return;
+      }
+
+      $.extend(true /*deep*/, item, response);
```
Comment (billyvg) [inline src/sentry/static/sentry/app/stores/teamStore.tsx:66]: 👀

File: src/sentry/static/sentry/app/stores/teamStore.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  },
+
+  onUpdateSuccess(itemId, response) {
+    if (!response) {
+      return;
+    }
+
+    const item = this.getBySlug(itemId);
+
+    if (!item) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/teamStore.tsx:51]: maybe we can remove "else" by using "return" here?

**Review by priscilawebdev — APPROVED** [review]
Great work! 👏I left just one suggestion😉

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/stores/teamStore.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/stores/teamStore.tsx:66]: I'll clean this up.

**Pull request merged**